templates/llm-router/data_utils.py

```diff
@@ -0,0 +1,178 @@
+import re
+import matplotlib.pyplot as plt
+from collections import Counter
+import pandas as pd
+
+
+def preprocess_nectar(df, model, model_name):
+    """
```

Comment:
Universal comment for all methods:
Always use type hinting (for both inputs and outputs) + docstring format should be:

```
"""<one line explanation>

<more detailed explanation>

Args:
     foo: explain foo
     bar: explain bar

Returns:
     explain returned items.
"""
```

Reply:
For this function in particular let's add more explanation in the docstring on what the preprocessing is doing? Like: Filter to include only the first turn, good-natured responses, and responses that contain the specified model